build/azDevops/azure/templates/steps/build-java.yml

```diff
@@ -146,15 +167,56 @@ steps:
 
   # SonarCloud
   - ${{ if eq(parameters.static_code_analysis, true) }}:
+      - bash: |
+          set -euxo pipefail
+          function strip_refs()
+          {
+            local BRANCH_REF="$1"
+
+            RETURN_BRANCH="$(sed -e "s%^refs/\(heads\|tags\)/%%" <<< $BRANCH_REF)"
+          }
+
+          if [ -z "${{ parameters.pullrequest_number }}" ]; then
+            strip_refs "${{ parameters.source_branch_ref }}"
+            EXTRA_SONAR_ARGUMENTS="-Dsonar.branch.name='$RETURN_BRANCH'"
+            strip_refs "${{ parameters.target_branch_ref }}"
+            EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} -Dsonar.branch.target='$RETURN_BRANCH'"
+          else
+            PROVIDER_LOWERCASED="$(tr '[:upper:]' '[:lower:]' <<< "${{ parameters.sonar_pullrequest_provider }}")"
+
+            EXTRA_SONAR_ARGUMENTS="-Dsonar.pullrequest.key='${{ parameters.pullrequest_number }}'"
+            strip_refs "${{ parameters.source_branch_ref }}"
+            EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} -Dsonar.pullrequest.branch='$RETURN_BRANCH'"
+            EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} -Dsonar.pullrequest.provider='$PROVIDER_LOWERCASED'"
+            strip_refs "${{ parameters.target_branch_ref }}"
+            EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} -Dsonar.pullrequest.base'$RETURN_BRANCH'"
+
+            if [ "$PROVIDER_LOWERCASED" == 'github' ]; then
+              EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} -Dsonar.pullrequest.github.repository='${{ parameters.sonar_remote_repo }}'"
+            fi
+          fi
+
+          echo "##vso[task.setvariable variable=bash_extra_sonar_arguments]$EXTRA_SONAR_ARGUMENTS"
+        displayName: "Sonar: Extra Sonar Arguments"
+
       - template: azDevOps/azure/templates/v2/steps/test-static-code-sonar.yml@templates
         parameters:
           working_directory: "${{ parameters.project_root_dir }}"
           target_container: "${{ parameters.sonar_container }}"
+          extra_arguments: "$(bash_extra_sonar_arguments)"
```

Comment:
why not make this EXTRA_SONAR_ARGS env_vars and default to ""? 
e.g. 
```env_vars: { 
   ... 
  EXTRA_SONAR_ARGS: "-Dsonar.pullrequest.github.repository='${{ parameters.sonar_remote_repo }}' -Dsonar.pullrequest.key='${{ parameters.pullrequest_number }}'  -Dsonar.pullrequest.branch='$RETURN_BRANCH' -Dsonar.pullrequest.provider='$PROVIDER_LOWERCASED'"
```

Reply:
I presume inside the function you are appending it after `sonar-scanner ${{ parameters.extra_arguments }}`  and defaulted the parameter to `""` instead you could also try this 
```
sonar-scanner ${EXTRA_SONAR_ARGS=""}
```

Reply:
Hmmm how so, does that work?

Reply:
that's a bash fallback value to "" in cases where users don't specify it. you could actually use ${EXTRA_SONAR_ARGS:=""}

Reply:
I can change it, but is there a reason to do it that way? I was thinking this way it's clear the arguments are appended to the command.

I don't mind either way.

Reply:
The other way you mentioned would pass it as an env var as well as extra arguments which might be unintended?